scripts/push-docker-image.sh

```diff
@@ -1,31 +1,71 @@
 #!/usr/bin/env bash
+#
+# Script is meant to build multi-arch container images and publish them to multiple container registries
+# 
+# Script is:
+# - figuring out if an image is a development one (by default suffixed with `-dev` in image name)
+# - figuring out the image tag (aka version) based on GITHUB_REF value
+#
+# Script is not:
+# - directly executing `docker build`. This is done in Makefile
+# - logging to registries 
+#
+
 # exit immediately when a command fails
 set -e
 # only exit with zero if all commands of the pipeline exit successfully
 set -o pipefail
-# error on unset variables
-set -u
 
 CPU_ARCHS="amd64 arm64 arm"
+REGISTRIES="${REGISTRIES:-"quay.io ghcr.io"}"
+
+# IMAGE_OPERATOR and IMAGER_RELOADER need to be exported to be used by `make`
+export IMAGE_OPERATOR="${IMAGE_OPERATOR:-"prometheus-operator/prometheus-operator"}"
+export IMAGE_RELOADER="${IMAGE_RELOADER:-"prometheus-operator/prometheus-config-reloader"}"
 
+# Figure out if current commit is tagged
 export TAG="${GITHUB_REF##*/}"
 
 # Push `-dev` images unless commit is tagged
-export REPO="${REPO:-"quay.io/prometheus-operator/prometheus-operator-dev"}"
-export REPO_PROMETHEUS_CONFIG_RELOADER="${REPO_PROMETHEUS_CONFIG_RELOADER:-"quay.io/prometheus-operator/prometheus-config-reloader-dev"}"
+IMAGE_SUFFIX="-dev"
```

Comment:
Does anyone actually use those `-dev` images? We don't list them anywhere in any docs so discovery is a bit hard. Also, those are build only from `master` branch and branches in the repo, so not really helping during standard development workflow. Maybe we should just remove them? @prometheus-operator/prometheus-operator-reviewers WDYT?

Reply:
No strong opinion but I can see how it can be helpful when you want to have users testing a bug fix or a change without having them to rebuild locally.

Reply:
I would agree if those images were built from PRs. As it is right now, the maintainer would need to create a branch to test the bug fix or a change and wait for CI to publish images. Personally, I would be in favor of simplifying image building process than keeping those `-dev` images.

Reply:
Fair enough. If nobody had ever used this possibility, I agree that it's not worth keeping it.